src/main/java/org/aerogear/kryptowire/BinaryHistoryAction.java

```diff
@@ -0,0 +1,106 @@
+package org.aerogear.kryptowire;
+
+import hudson.Extension;
+import hudson.FilePath;
+import hudson.model.*;
+import jenkins.model.GlobalConfiguration;
+import jenkins.model.Jenkins;
+import jenkins.model.RunAction2;
+import org.aerogear.kryptowire.BinaryInfo;
+import org.aerogear.kryptowire.GlobalConfigurationImpl;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.springframework.ejb.config.JeeNamespaceHandler;
+
+import java.io.IOException;
+import java.util.Properties;
+
+
+public class BinaryHistoryAction implements RunAction2 {
+    private BinaryInfo info;
+    private transient Run run;
+    private BinaryStatus status;
+    private String externalLink;
+
+
+    public BinaryHistoryAction(BinaryInfo info) {
+        this.info = info;
+    }
+
+    public BinaryHistoryAction(BinaryInfo info, BinaryStatus status) {
+        this.info = info;
+        this.status = status;
+    }
+
+    @Override
+    public void onAttached(Run<?, ?> run) {
+        this.run = run;
+    }
+
+    @Override
+    public void onLoad(Run<?, ?> run) {
+        this.run = run;
+    }
+
+    @Override
+    public String getIconFileName() {
+        return "/plugin/aerogear-kryptowire-plugin/menu-logo.png";
+    }
+
+    @Override
+    public String getDisplayName() {
+        return "Kryptowire Scan Results";
+    }
+
+    @Override
+    public String getUrlName() {
+        return "kryptowire";
+    }
+
+    public BinaryInfo getInfo() {
+        return info;
+    }
+
+    public void setInfo(BinaryInfo info) {
+        this.info = info;
+    }
+
+    public Run getRun() {
+        return run;
+    }
+
+    public void setRun(Run run) {
+        this.run = run;
+    }
+
+
+    public void setStatus(BinaryStatus status) {
+        this.status = status;
+    }
+
+    public BinaryStatus getStatus() throws IOException, InterruptedException {
+        if (this.status != null) {
+            return this.status;
+        }
+        GlobalConfigurationImpl cfg = GlobalConfiguration.all().get(GlobalConfigurationImpl.class);
```

Comment:
I had to remove the global config as a instance property (it is being used in another method) because both kryptowire endpoint and api key were being stored into the build.xml job file.